src/Symfony/Component/AssetMapper/ImportMap/RemotePackageDownloader.php

```diff
@@ -0,0 +1,161 @@
+
+            $remoteEntriesToDownload[$entry->importName] = $entry;
+        }
+
+        if (0 === \count($remoteEntriesToDownload)) {
+            return [];
+        }
+
+        $contents = $this->packageResolver->downloadPackages($remoteEntriesToDownload, $progressCallback);
+        $downloadedPackages = [];
+        foreach ($contents as $package => $content) {
+            if (!isset($remoteEntriesToDownload[$package])) {
+                throw new \LogicException(sprintf('The package "%s" was not in the requested packages.', $package));
+            }
+
+            $filename = $this->savePackage(
+                $package,
+                $content,
+                $remoteEntriesToDownload[$package]->type,
+            );
+            $newInstalled[$package] = [
+                'path' => $filename,
+                'version' => $remoteEntriesToDownload[$package]->version,
+            ];
+
+            $downloadedPackages[] = $package;
+        }
+
+        $this->saveInstalled($newInstalled);
+
+        return $downloadedPackages;
+    }
+
+    public function getDownloadedPath(string $importName): string
+    {
+        $installed = $this->loadInstalled();
+        if (!isset($installed[$importName])) {
+            throw new \InvalidArgumentException(sprintf('The "%s" vendor asset is missing. Run "php bin/console importmap:install".', $importName));
+        }
+
+        return $this->vendorDir.'/'.$installed[$importName]['path'];
+    }
+
+    public function getVendorDir(): string
+    {
+        return $this->vendorDir;
+    }
+
+    private function savePackage(string $packageName, string $packageContents, ImportMapType $importMapType): string
+    {
+        $filename = $packageName;
+        if (!str_contains(basename($packageName), '.')) {
+            $filename .= '.'.$importMapType->value;
+        }
+        $vendorPath = $this->vendorDir.'/'.$filename;
+
+        @mkdir(\dirname($vendorPath), 0777, true);
+        file_put_contents($vendorPath, $packageContents);
+
+        return $filename;
+    }
+
+    /**
+     * @return array<string, array{path: string, version: string}>
+     */
+    private function loadInstalled(): array
+    {
+        if (isset($this->installed)) {
+            return $this->installed;
+        }
+
+        $installedPath = $this->vendorDir.'/installed.php';
+        if (!file_exists($installedPath)) {
+            return [];
+        }
+
+        $installed = is_file($installedPath) ? (static fn () => include $installedPath)() : [];
+
+        foreach ($installed as $package => $data) {
+            if (!isset($data['path'])) {
+                throw new \LogicException(sprintf('The package "%s" is missing its path.', $package));
+            }
+
+            if (!isset($data['version'])) {
+                throw new \LogicException(sprintf('The package "%s" is missing its version.', $package));
+            }
+
+            if (!file_exists($this->vendorDir.'/'.$data['path'])) {
```

Comment:
we try to use is_file instead of file_exists as much as possible (it's faster)
```suggestion
            if (!is_file($this->vendorDir.'/'.$data['path'])) {
```